From 13c6a04367ed053ce6b841512b008adc9c3a52a4 Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jpirko@redhat.com>
Date: Thu, 7 Aug 2014 07:06:25 -0400
Subject: [net] ip_tunnel: Remove double unregister of the fallback device

Message-id: <1407395194-3809-10-git-send-email-jpirko@redhat.com>
Patchwork-id: 87523
O-Subject: [patch rhel7 net 09/18] ip_tunnel: Remove double unregister of the fallback device
Bugzilla: 1091561
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

BZ1091561
https://brewweb.devel.redhat.com/taskinfo?taskID=7794261
Tested by jtluka's LNST smoke test recipe

upstream commit cfe4a536927c3186b7e7f9be688e7e0f62bb8ea1
Author: Steffen Klassert <steffen.klassert@secunet.com>
Date:   Tue Oct 1 11:37:37 2013 +0200

    ip_tunnel: Remove double unregister of the fallback device

    When queueing the netdevices for removal, we queue the
    fallback device twice in ip_tunnel_destroy(). The first
    time when we queue all netdevices in the namespace and
    then again explicitly. Fix this by removing the explicit
    queueing of the fallback device.

    Bug was introduced when network namespace support was added
    with commit 6c742e714d8 ("ipip: add x-netns support").

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/ip_tunnel.c b/net/ipv4/ip_tunnel.c
index 0f5c42a..dddab56 100644
--- a/net/ipv4/ip_tunnel.c
+++ b/net/ipv4/ip_tunnel.c
@@ -903,8 +903,6 @@ static void ip_tunnel_destroy(struct ip_tunnel_net *itn, struct list_head *head,
    if (!net_eq(dev_net(t->dev), net))
     unregister_netdevice_queue(t->dev, head);
  }
- if (itn->fb_tunnel_dev)
-  unregister_netdevice_queue(itn->fb_tunnel_dev, head);
 }
 
 void ip_tunnel_delete_net(struct ip_tunnel_net *itn, struct rtnl_link_ops *ops)
-- 
1.7.1